### DISCUSSION TITLE
cake.tool 1.0.0-rc0001 and cake.bakery VSCode IntelliSense System.Runtime 5.0.0.0 missing assembly reference

### DISCUSSION THREAD

**Discussion by jimalad** [cake-build/cake#3001] [Q&A]
OS: Windows 10 x64
VSCode: 1.52.1
VSCode cake extension: 1.0.0
dotnet-sdk: 3.1.404-win-x64
dotnet tool list --global: cake.tool 1.0.0-rc0001
ctrl-shift-p > cake > Install sample build file > build.cake
ctrl-shift-p > cake > Install intellisense support

Then after a while (30 seconds or so) / maybe including a VSCode restart...

**Outcome**: red squigglies / problems throughout cake source e.g.
The type 'Object' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. [build.cake]

Searched for ages, found this https://github.com/dotnet/roslyn/issues/49498 but not sure it's relevant.

BTW, is what I'm describing above the best practice for cake development in VSCode with intellisense?
I watched the video linked here https://cakebuild.net/blog/2020/12/cake-vscode-extension-release-1.0.0 and noticed IntelliSense wasn't demonstrated. I realise Cake addon for VSCode and bakery are different things, but would think the UX would have been more unified. I rarely see anything discussed in issues about IntelliSense - is that because it's working great for everyone, or that it's rarely used?

**Comment by gep13** [comment]
@jimalad sorry to hear that you are having issues!  I just wanted to mention that the focus of the video that you referenced was the new features that were shipping as part of the 1.0.0 release of the Cake Extension for VSCode.  Intellisense has been around for a while, at it shouldn’t have been impacted by any of the new features.  You can see intellisense in action in a recent talk that I gave at .NET Conf here: https://youtu.be/o38glTSZsVg.

Now, that isn’t to say that there isn’t a problem though.

I believe I am right in saying that @mholo65 is going to follow up requesting some more detailed logging for the issue that you are seeing, and that will hopefully help to get to the bottom of the problem.

**Reply by pascalberger** [reply]
There are also some troubleshooting tips (including how to get OmniSharp logs), which you can try out in the meantime, available here: https://cakebuild.net/docs/integrations/editors/vscode/intellisense#troubleshooting.

**Reply by jimalad** [reply]
Thanks. Big fan of cake, thank you for all your efforts.

I re-reviewed the troubleshooting link. I'd been there earlier but hadn't spotted that there are four UI tabs, one of each type of runner.
In the 'Cake runner for .NET Core' tab, I noticed "Make sure that Cake.CoreCLR is installed in your tools folder". This gave me something to try out.



With VSCode settings; Omnisharp logging level set to trace...


**Test 1**
With cake 0.38.5 installed as dotnet global tool
With "c:\my\work\space\Build\tools\Cake.Bakery\tools" containing *only* "Cake.Bakery" subdirectory

```
[info]: OmniSharp.Cake.Services.CakeScriptService
        Using Cake.Bakery at c:\my\work\space\Build\tools\Cake.Bakery\tools\Cake.Bakery.exe
[trce]: OmniSharp.Services.AssemblyLoader
        Assembly loaded from path: C:/Users/user/.dotnet/tools/.store/cake.tool/0.38.5/cake.tool/0.38.5/tools/netcoreapp2.1/any/Cake.Core.dll
```

Outcome: IntelliSense working correctly


**Test 2**
With cake 1.0.0-rc0001 installed as dotnet global tool
With "c:\my\work\space\Build\tools\Cake.Bakery\tools" containing *only* "Cake.Bakery" subdirectory

```
[info]: OmniSharp.Cake.Services.CakeScriptService
        Using Cake.Bakery at c:\my\work\space\Build\tools\Cake.Bakery\tools\Cake.Bakery.exe
[trce]: OmniSharp.Services.AssemblyLoader
        Assembly loaded from path: C:/Users/user/.dotnet/tools/.store/cake.tool/1.0.0-rc0001/cake.tool/1.0.0-rc0001/tools/net5.0/any/Cake.Core.dll
```

Outcome: IntelliSense not working correctly. Omnisharp log littered with stuff like...
```
{
  "LogLevel": "Error",
  "Id": "CS0012",
  "Tags": [],
  "FileName": "c:/my/work/space/Build/build.cake",
  "Line": 34,
  "Column": 2,
  "EndLine": 34,
  "EndColumn": 6,
  "Text": "The type 'Object' is defined in an assembly that is not referenced. You must add a reference to assembly 'System.Runtime, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.",
  "Projects": [
    "build.cake"
  ]
}
```


**Test 3**
With cake 1.0.0-rc0001 installed as dotnet global tool
With "c:\my\work\space\Build\tools\Cake.Bakery\tools" containing *both* "Cake.Bakery" and "Cake.CoreCLR.1.0.0-rc0001" subdirectories

```
[info]: OmniSharp.Cake.Services.CakeScriptService
        Using Cake.Bakery at c:\my\work\space\Build\tools\Cake.Bakery\tools\Cake.Bakery.exe
[trce]: OmniSharp.Services.AssemblyLoader
        Assembly loaded from path: c:/my/work/space/Build/tools/Cake.CoreCLR.1.0.0-rc0001/Cake.Core.dll
```

Outcome: IntelliSense working correctly



So I guess my question is: Why in Test 2 does it find Cake.Core.dll in .dotnet/tools, but it doesn't seem to be happy with it like it was for 0.38.5 in Test 1?

**Reply by bjorkstromm** [reply]
What is the output of the following commands, from the root of your project?

```
dotnet --version
```

```
dotnet --list-sdks
```

My guess is that when 1.0.0-rcxx version is used, then Bakery picks up a .NET5 version of Cake.Core.dll and Cake.Common.dll.

**Reply by jimalad** [reply]
I've just tried both of the following scenarios by uninstalling / reinstalling a single installed version of .NET Core SDK on the test VM I'm working with.

```
dotnet --version
3.1.404

dotnet --list-sdks
3.1.404 [C:\Program Files\dotnet\sdk]
```

and

```
dotnet --version
5.0.101

dotnet --list-sdks
5.0.101 [C:\Program Files\dotnet\sdk]
```

In both of these SDK scenarios, I get the same results for Test 2 and Test 3 (above).
